clippy_lints/src/utils/internal_lints.rs

```diff
@@ -207,3 +220,34 @@ impl<'a, 'tcx: 'a> Visitor<'tcx> for LintCollector<'a, 'tcx> {
         NestedVisitorMap::All(&self.cx.tcx.hir)
     }
 }
+
+
+
+pub struct DefaultHashTypes {
+    map: FxHashMap<String, String>,
+}
+
+impl DefaultHashTypes {
+    pub fn new() -> Self {
+        let mut map = FxHashMap::default();
+        map.insert("HashMap".to_owned(), "FxHashMap".to_owned());
+        map.insert("HashSet".to_owned(), "FxHashSet".to_owned());
+        Self { map }
+    }
+}
+
+impl LintPass for DefaultHashTypes {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(DEFAULT_HASH_TYPES)
+    }
+}
+
+impl EarlyLintPass for DefaultHashTypes {
+    fn check_ident(&mut self, cx: &EarlyContext<'_>, ident: Ident) {
+        let ident_string = ident.to_string();
+        if let Some(replace) = self.map.get(&ident_string) {
+            let msg = format!("Prefer {} over {}, it has better performance and we don't need any collision prevention in clippy", replace, ident_string);
+            cx.span_lint(DEFAULT_HASH_TYPES, ident.span, &msg);
```

Comment:
I think using the `utils::span_lint_and_sugg` function here would be even nicer.
https://github.com/rust-lang-nursery/rust-clippy/blob/e6d92f95c6d6964628d75c25fb3d1a58a9f25e5d/clippy_lints/src/utils/mod.rs#L580-L587
`help`: `"use"`
`sugg`: `replace`

You should always use the `span_lint` functions from the `utils` module for Clippy lints, because they add the link to the doc to the warning messages.

Sorry I missed that in the previous review.